### PR TITLE
Include zap comments in artist results

### PR DESCRIPTION
**Pull request by blastshielddown**
For #145

### CODE REVIEW

File: controllers/comments.ts [hunk 1]
```diff
@@ -132,11 +62,38 @@ const get_artist_comments = asyncHandler(async (req, res, next) => {
...
+    .min("user.artwork_url as commenterArtworkUrl")
+    .where("artist.id", "=", artistId)
+    .andWhere("amp.comment", "=", true)
+    .andWhere("track.deleted", "=", false)
+    .whereNull("comment.parent_id")
+    .groupBy("comment.id", "track.id", "amp.tx_id")
+    .orderBy("createdAt", "desc");
+
+  // TODO: Pagination
+  res.json({ success: true, data: [...comments, ...commentsLegacy] });
```
Comment (joshr4) [inline controllers/comments.ts:96]: Does commentsLegacy only include comments before a certain date? Or will this ever return recent comments? Wondering if we need to sort this merged array by date or not.

File: library/comments.ts [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    }),
+  ]);
+
+  const commentsWithUserInfo = await Promise.all(
+    userComments.map(async (comment) => {
+      if (comment.userId === "keysend") {
+        return comment;
+      }
+
+      const user = await prisma.user.findUnique({
```
Comment (joshr4) [inline library/comments.ts:39]: Could this be done in a join with the non nostr comment query above? Or are we limited by the how Prisma relates these two tables, comment and user? I maybe recall some table relationship limitation here...

Comment (blastshielddown) [reply library/comments.ts:39]: yeah we would need to create a relation to get the results in a single query

File: controllers/comments.ts (hunk 1 above)
Comment (blastshielddown) [reply controllers/comments.ts:96]: Checking on this as I'm not sure. Also the way it's written isn't returning all comments as I expected. Need to fix this

Comment (blastshielddown) [reply controllers/comments.ts]: Updated to run legacy query inside the library function, then sort and return results

**Review by joshr4 — APPROVED** [review]

**Pull request merged**